packages/core/src/component.ts

```diff
@@ -3,34 +3,52 @@
 // This file is licensed under the MIT License.
 // License text available at https://opensource.org/licenses/MIT
 
-import {Constructor, Provider, BoundValue} from '@loopback/context';
-import {Application} from '.';
-
-// tslint:disable:no-any
+import { Constructor, Provider, BoundValue, Context } from '@loopback/context';
+import {Server} from '.';
 
 export interface ProviderMap {
   [key: string]: Constructor<Provider<BoundValue>>;
 }
 
+export interface ControllerMap {
+  [key: string]: Constructor<{}>;
+}
+
 export interface Component {
-  controllers?: Constructor<any>[];
+  controllers?: ControllerMap;
```

Comment:
If controllers are transport specific, then the base component interface should not contain controllers config and this config should be contributed by the extension. Consistently with the approach we use for repositories - see #541.

Reply:
Based on subsequent refactoring, we'll keep controllers as application-level items, so this has been reverted.